SIPS/sip-58.md

```diff
@@ -0,0 +1,80 @@
+
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the SIP.-->
+
+Emit list of individual fee reclamations and rebates during exchange settlement for Dapps and Synthetix Exchange. Fee reclamation was introduced since [SIP-37](./sip-37.md) but did not emit a breakdown of each individual trade's fee reclamation / rebate amounts.
+
+## Abstract
+
+<!--A short (~200 word) description of the technical issue being addressed.-->
+
+Upgrade the Exchanger `settle()` function to emit individual fee reclaimation / rebate amounts for each trade.
+
+Add an internal function `_settlementsOwing` that will emit an event `ExchangeEntriesSettled` when `Exchanger.settle()` is invoked.
+
+Emit a list for each exchange entry with details about the `amountShouldHaveReceived`, the corresponding `amountReceived`, `reclaimAmount`, `rebateAmount`, and `timestamp` of the entry that is settled in the transaction.
+
+## Motivation
+
+<!--The motivation is critical for SIPs that want to change Synthetix. It should clearly explain why the existing protocol specification is inadequate to address the problem that the SIP solves. SIP submissions without sufficient motivation may be rejected outright.-->
+
+We want to display on Synthetix exchange the corresponding fee reclamation amounts (if any) for each individual exchange made when exchange settlement is invoked.
+
+Currently invoking `settle()` will only emit an event, if any, with the total sum of any fee reclamation or rebate amounts.
+
+## Specification
+
+<!--The technical specification should describe the syntax and semantics of any new feature.-->
+
+### Exchanger.settle() ###
+
+Add an internal function `_settlementsOwing` that will emit an event `ExchangeEntriesSettled` when `Exchanger.settle()` is invoked.
+
+`Exchanger.settle()` will use `_settlementsOwing` to calculate the fee reclamation amounts.
+
+**Event**
+
+Emit an event `ExchangeEntriesSettled` when `Exchanger.settle()` is invoked within `_settlementsOwing`.
+
+```solidity
+event ExchangeEntriesSettled(address indexed from, bytes32 currencyKey, uint numEntries, uint[] amountReceived, uint[] amountShouldHaveReceived, uint[] reclaim, uint[] rebate, uint[] timestamp);
+```
+
+## Rationale
+
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+
+It is important that traders can see on each trade the fee reclamation and rebates that occur after they make another exchange that invokes `settle()`, for calculating trading profits and loses based on the amount and effective price they recieved on each individual trade.
```

Comment:
Can you explain your reasoning of choices in the spec - why create another `settlementOwing` version? Etc. (Otherwise it just reads like the Motivation again)